crates/ruff_linter/src/rules/pylint/settings.rs

```diff
@@ -17,19 +16,19 @@ pub enum ConstantType {
     Str,
 }
 
-impl TryFrom<&Constant> for ConstantType {
-    type Error = anyhow::Error;
+impl TryFrom<&Expr> for ConstantType {
+    type Error = ();
 
-    fn try_from(value: &Constant) -> Result<Self, Self::Error> {
+    fn try_from(value: &Expr) -> Result<Self, Self::Error> {
         match value {
-            Constant::Bytes(..) => Ok(Self::Bytes),
-            Constant::Complex { .. } => Ok(Self::Complex),
-            Constant::Float(..) => Ok(Self::Float),
-            Constant::Int(..) => Ok(Self::Int),
-            Constant::Str(..) => Ok(Self::Str),
-            Constant::Bool(..) | Constant::Ellipsis | Constant::None => {
-                Err(anyhow!("Singleton constants are unsupported"))
-            }
+            Expr::StringLiteral(_) => Ok(Self::Str),
+            Expr::BytesLiteral(_) => Ok(Self::Bytes),
+            Expr::NumberLiteral(ExprNumberLiteral { value, .. }) => match value {
+                Number::Int(_) => Ok(Self::Int),
+                Number::Float(_) => Ok(Self::Float),
+                Number::Complex { .. } => Ok(Self::Complex),
+            },
+            _ => Err(()),
```

Comment:
The `Err` variant was never used and this won't be an error. On that note, I think this could be better as it's own method return `Option<Self>` instead.

Reply:
It's not uncommon for `new` or `from` methods to return `Option<Self>`. Do you want to change it?